spock-core/src/main/java/org/spockframework/runtime/extension/builtin/TempDirExtension.java

```diff
@@ -29,21 +43,47 @@ public TempDirExtension(TempDirConfiguration configuration) {
 
   @Override
   public void visitFieldAnnotation(TempDir annotation, FieldInfo field) {
-    TempDirInterceptor interceptor = TempDirInterceptor.forField(field, configuration.baseDir, configuration.keep);
+    TempDir.CleanupMode cleanupMode = annotation.cleanup();
+    cleanupMode = cleanupMode == TempDir.CleanupMode.DEFAULT ? configuration.cleanup : cleanupMode;
```

Comment:
How about extracting these two lines in a helper method, because the same logic is also used in `visitParameterAnnotation` which should behave the same.